packages/react-basemaps/src/basemaps/GoogleMap.js

```diff
@@ -120,12 +121,16 @@ export function GoogleMap(props) {
         window.cartoGmap.setZoom(zoom);
       }
       window.cartoGmap.setOptions(rest);
-      window.cartoDeck.setProps({ layers, getTooltip });
+      window.cartoDeck.setProps({ getTooltip });
     }
-
-    window.cartoDeck.setMap(window.cartoGmap);
   };
 
+  // Set layers should be outside of the useEffect to avoid problems with layers that changes
```

Comment:
Won't it cause useless redraws on Deck?

Also, is it normal that the `useEffect` below has no dependency array?

Reply:
`Won't it cause useless redraws on Deck?`

No, in fact, Deck always should receive a new instance of all layers and it decides if it's necessary to redraw or not. The creation of the layers or setting them within a `useEffect` is a common mistake. Check this example https://deck.gl/docs/get-started/using-with-react#the-deckgl-react-component

`Also, is it normal that the useEffect below has no dependency array?`
In this case yes although I admit that it is difficult to understand. In any case, this PR does not intend to modify that part of the code